extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -150,6 +195,7 @@ export class AmpAd3PImpl extends AMP.BaseElement {
    * @override
    */
   onLayoutMeasure() {
+    this.lifecycleReporter_.sendPing('adRequestStart');
```

Comment:
regardingless to the implementations of 3p and a4a, i think probably we're interested in those time points:

- adslotBuilt: when the DOM element is parsed
- adRequestStart: when the ad request is sent (URL is built before that)
- adRenderStart: when ad response is back and it has ad
- adNoFill: when ad response is back but it has no ad

All the naming are TBD, the idea is we have a comparable common path for a4a and 3p. Of course, a4a could add other pings in between those ones for a4a specific things.

Addition to that, we should be able to add property to the pings to identify if it's a4a or 3p, so we can have nice chart break down to compare the traffic side by side. If that's something currently missing, I'm OK to use name prefix/suffix.